src/lib/blkdata_svc/blkdata_service.cpp

```diff
@@ -13,95 +13,123 @@
  * specific language governing permissions and limitations under the License.
  *
  *********************************************************************************/
-#include "../api/blkdata_service.hpp"
-#include "../device/virtual_dev.hpp"
-#include "../homestore.hpp"
+#include <homestore/blkdata_service.hpp>
+#include <homestore/homestore.hpp>
+#include "device/virtual_dev.hpp"
+#include "device/physical_dev.hpp"     // vdev_info_block
+#include "common/homestore_config.hpp" // is_data_drive_hdd
+#include "common/error.h"
+#include "blk_read_tracker.hpp"
 
 namespace homestore {
-BlkDataService::BlkDataService(uint64_t size, uint32_t page_size, blk_allocator_type_t blkalloc_type,
-                               bool cache = false) {
-    HS_REL_ASSERT_EQ(cache, false, "We don't support cached blkdata service yet");
-
-    struct blkstore_blob hdr {};
-    hdr.type = blkstore_type::DATA_STORE;
-
-    m_vdev = std::make_unique< VirtualDev >(hs()->device_manager(), "DataVDev", PhysicalDevGroup::DATA, blkalloc_type,
-                                            sizeof(blkstore_blob), 0, true /* is_stripe */, page_size,
-                                            bind_this(BlkDataService::vdev_io_completion, 1), hdr, size,
-                                            true /* auto_recovery */);
-    m_page_size = page_size;
+
+BlkDataService::BlkDataService() { m_blk_read_tracker = std::make_unique< BlkReadTracker >(); }
+BlkDataService::~BlkDataService() {}
```

Comment:
You could declare it .h file as `~BlkDataService() = default`

Reply:
fixed